Data/Primitive/UnliftedArray.hs

```diff
@@ -137,6 +141,22 @@ instance PrimUnlifted (MV.MutVar s a) where
   toArrayArray# (MV.MutVar mv#) = unsafeCoerce# mv#
   fromArrayArray# aa# = MV.MutVar (unsafeCoerce# aa#)
 
+instance PrimUnlifted (MVar a) where
+  toArrayArray# (MVar mv#) = unsafeCoerce# mv#
```

Comment:
To my understanding, `ArrayArray#` can hold any values whose type is of kind `TYPE UnliftedRep`. This seems safe to me, but it would be nice to have that confirmed by someone else. @bgamari Can you comment on this?